projects/igniteui-angular/src/lib/grids/grid-base.component.ts

```diff
@@ -102,6 +101,11 @@ const MIN_ROW_EDITING_COUNT_THRESHOLD = 2;
 
 export const IgxGridTransaction = new InjectionToken<string>('IgxGridTransaction');
 
+export let warningShown = false;
```

Comment:
You can name these two variable more accurate in order to be clear for which warning they apply. Imagine that someone adds one more deprecation warning.